cpp/src/parquet/statistics_test.cc

```diff
@@ -602,18 +602,25 @@ class TestStatisticsHasFlag : public TestStatistics<TestType> {
   }
 
   std::shared_ptr<TypedStatistics<TestType>> MergedStatistics(
-      const TypedStatistics<TestType>& stats1, const TypedStatistics<TestType>& stats2) {
-    auto chunk_statistics = MakeStatistics<TestType>(this->schema_.Column(0));
+      const TypedStatistics<TestType>& stats1, const TypedStatistics<TestType>& stats2,
+      const EncodedStatistics* initial_statistics) {
+    auto chunk_statistics = std::static_pointer_cast<TypedStatistics<TestType>>(
+        initial_statistics != nullptr
+            ? Statistics::Make(this->schema_.Column(0), initial_statistics)
+            : Statistics::Make(this->schema_.Column(0)));
     chunk_statistics->Merge(stats1);
     chunk_statistics->Merge(stats2);
     return chunk_statistics;
   }
 
   void VerifyMergedStatistics(
       const TypedStatistics<TestType>& stats1, const TypedStatistics<TestType>& stats2,
-      const std::function<void(TypedStatistics<TestType>*)>& test_fn) {
-    ASSERT_NO_FATAL_FAILURE(test_fn(MergedStatistics(stats1, stats2).get()));
-    ASSERT_NO_FATAL_FAILURE(test_fn(MergedStatistics(stats2, stats1).get()));
+      const std::function<void(TypedStatistics<TestType>*)>& test_fn,
+      const EncodedStatistics* initial_statistics = nullptr) {
```

Comment:
The signature is a bit weird here. How about two different overloads?
```c++

  void VerifyMergedStatistics(
      const TypedStatistics<TestType>& stats1, const TypedStatistics<TestType>& stats2,
      const std::function<void(TypedStatistics<TestType>*)>& test_fn) { ... }
  void VerifyMergedStatistics(
      const EncodedStatistics& initial_statistics,
      const TypedStatistics<TestType>& stats1, const TypedStatistics<TestType>& stats2,
      const std::function<void(TypedStatistics<TestType>*)>& test_fn) { ... }
```

Reply:
Now we just use `MergeDistinctCount` for the distinct count tests, so I removed the overload.